entity-framework/core/providers/cosmos/querying.md

```diff
@@ -0,0 +1,312 @@
+```
+
+Note the `ReadItem`, which indicates that the query was executed as an efficient point read - no SQL query is involved.
+
+Note that as with partition key extraction, significant improvements have been made to this mechanism in EF 9.0; older versions do not reliably detect and use point reads.
+
+## Pagination
+
+> [!NOTE]
+> This feature was introduced in EF Core 9.0.
+
+Pagination refers to retrieving results in pages, rather than all at once; this is typically done for large resultsets, where a user interface is displayed, allowing users to navigate through pages of the results.
+
+A common way to implement pagination with databases is to use the `Skip` and `Take` LINQ operators (`OFFSET` and `LIMIT` in SQL). Given a page size of 10 results, the third page can be fetched with EF Core as follows:
+
+```csharp
+var position = 20;
+var nextPage = context.Session
+    .OrderBy(s => s.Id)
+    .Skip(position)
+    .Take(10)
+    .ToList();
+```
+
+Unfortunately, this technique is quite inefficient and can considerably increase querying costs. Cosmos DB provides a special mechanism for paginating through the result of a query, via the use of _continuation tokens_:
+
+```csharp
+CosmosPage firstPage = await context.Sessions
+    .OrderBy(s => s.Id)
+    .ToPageAsync(pageSize: 10, continuationToken: null);
+
+string continuationToken = page.ContinuationToken;
+foreach (var session in firstPage.Values)
+{
+    // Display/send the sessions to the user
+}
+```
+
+Rather than terminating the LINQ query with `ToListAsync` or similar, we use the `ToPageAsync` method, instructing it to get at most 10 items in every page (note that there may be fewer items in the database). Since this is our first query, we'd like to get results from the beginning, and pass `null` as the continuation token. `ToPageAsync` returns a `CosmosPage`, which exposes a continuation token and the values in the page (up to 10 items). Your program will typically send those values to the client, along with the continuation token; this will allow resuming the query later and fetching more results.
+
+Let's assume the user now clicks on the "Next" button in their UI, asking for the next 10 items. You can then execute the query as follows:
+
+```csharp
+CosmosPage nextPage = await context.Sessions.OrderBy(s => s.Id).ToPageAsync(10, continuationToken);
+string continuationToken = nextPage.ContinuationToken;
+foreach (var session in nextPage.Values)
+{
+    // Display/send the sessions to the user
+}
+```
+
+We execute the same query, but this time we pass in the continuation token received from the first execution; this instructs Cosmos DB to continue the query where it left off, and fetch the next 10 items. This method of paginating is extremely efficient and cost-effective compared to using `Skip` and `Take`.
```

Comment:
Consider mentioning that `continuationToken` will be `null` if there are no more results